### PR TITLE
Task 05

### PR DESCRIPTION
**Pull request by jake4369**
5. GET /api/reviews/:review_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -66,6 +66,40 @@ describe("GET /api/reviews", () => {
...
+
+    return request(app)
+      .get("/api/reviews/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { review } = body;
+        expect(review).toMatchObject(expectedReview);
+      });
+  });
+  it("should respond with a 204 status code if no review is found", () => {
```
Comment (HannahBurks) [inline __tests__/app.test.js:92]: The status code in this case would be 404 not 204 - if the it is a non existent ID

File: __tests__/app.test.js [hunk 2]
```diff
@@ -66,6 +66,40 @@ describe("GET /api/reviews", () => {
...
+    return request(app)
+      .get("/api/reviews/100")
+      .expect(204)
+      .then(({ body }) => {
+        const { message } = body;
+        expect(message).toBe(undefined);
+      });
+  });
+});
+
```
Comment (HannahBurks) [inline __tests__/app.test.js:102]: We should also have a 400 test and error handling for if it is an invalid ie (ie not a number like /monkey)

File: controllers/reviewsController.js [hunk 3]
```diff
@@ -10,3 +10,19 @@ exports.getReviews = (req, res, next) => {
...
     .catch((error) => next(error));
 };
+
+exports.getReview = (req, res, next) => {
+  const { review_id } = req.params;
+
+  reviewsModel
+    .getReview(review_id)
+    .then((review) => {
+      if (!review) {
```
Comment (HannahBurks) [inline controllers/reviewsController.js:20]: We can remove this 204 handler and instead if the review does not exist - we can check whether it is due to invalid ID number or whether the ID is not a number (like a string) and depending on that, we can then invoke the correct error handler (400 or 404) and create an error handler for those in an error.js file

File: models/reviewsModel.js [hunk 4]
```diff
@@ -16,3 +16,18 @@ exports.getReviews = () => {
...
+exports.getReview = (review_id) => {
+  return db
+    .query(
+      `
+      SELECT * FROM reviews
+      WHERE review_id = $1
+    `,
+      [review_id]
+    )
+    .then((result) => {
```
Comment (HannahBurks) [inline models/reviewsModel.js:29]: Here we can add a custom error to handle the 404 - check length of result and if there is nothing we know it is a 404

File: __tests__/app.test.js (hunk 1 above)
Comment (cjplatten) [inline __tests__/app.test.js:92]: Statuses in the 200s imply the request was successful, "review not found" should return a different error code.

**Review by HannahBurks — CHANGES_REQUESTED** [review]
I would now suggest adding an error handling file to deal with customErrors and PSQLerrors. 
This means if there is an error code or a promise.reject message is thrown in the model, it will invoke the error handling file and be dealt with accordingly. Once these error handlers have been added and tests for those errors, please submit another PR

File: __tests__/app.test.js [hunk 5]
```diff
@@ -66,6 +66,40 @@ describe("GET /api/reviews", () => {
...
 describe("404 error on /api/not-path", () => {
   it("status 404 returns error message bad path when provided an invalid path", () => {
```
Comment (cjplatten) [inline __tests__/app.test.js:104]: 404 is for if the input is valid but the item cannot be found, a bad client request has a different error code.

**Review by HannahBurks — APPROVED** [review]
Amazing - merge ahead Jake !😃

**Pull request merged**